src/services/pin-state-transition/index.js

```diff
@@ -0,0 +1,136 @@
+const errors = require('feathers-errors');
+const hooks = require('./hooks');
+const Pin = require('../pin/pin-model');
+const roles = require('../../constants/roles');
+const states = require('../../constants/pin-states');
+
+// States
+const UNVERIFIED = states.UNVERIFIED;
+const VERIFIED = states.VERIFIED;
+const ASSIGNED = states.ASSIGNED;
+const PROCESSING = states.PROCESSING;
+const RESOLVED = states.RESOLVED;
+const REJECTED = states.REJECTED;
+
+// Roles
+const SUPER_ADMIN = roles.SUPER_ADMIN;
+const ORGANIZATION_ADMIN = roles.ORGANIZATION_ADMIN;
+const DEPARTMENT_HEAD = roles.DEPARTMENT_HEAD;
+const USER = roles.USER;
+
+class PinTransitionService {
+
+  static isValidStateTransition(prevState, nextState, role) {
+    // Map previous state and role to a list of possible next states
+    const possibleNextStates = {
```

Comment:
I see. I tried to map with the RESTful API that specifies **what** the shape of resource we want it to be, not **how**. I still feel uncomfortable with it either or I don't fully understand it yet. Let it be for now.